### ISSUE TITLE
SASS loader with extract text Plugin not processing the SCSS file to CSS

### ISSUE THREAD

**Issue by buddyackerman** [webpack-contrib/extract-text-webpack-plugin#778]
I'm using webpack version 1.13.2 and sass-loader 4.1.1 and Extract Text Plugin 1.01.  When I use the following syntax the webpack build completes and outputs a .css file but it's nothing more than the original file still in SCSS format, not CSS

    module: {
        loaders: [
            { test: /\.scss$/, loader: ExtractTextPlugin.extract('style-loader', 'css-loader', 'sass-loader') },
        ]
    },

When I use the below syntax I get an exception thrown (the list of loaders is in brackets):

    module: {
        loaders: [
            { test: /\.scss$/, loader: ExtractTextPlugin.extract(['style-loader', 'css-loader', 'sass-loader']) },
        ]
    },

Here's the error:

    ERROR in ./assets/navigation.scss
    Module build failed: ReferenceError: self is not defined
        at C:\Projects\Client\node_modules\style-loader\index.js!C:\Projects\Client\node_modules\css-loader\index.js!C:\Projects\Client\node_modules\sass-loader\index.js!C:\Projects\Client\assets\navigation.scss:157:31
        at C:\Projects\Client\node_modules\style-loader\index.js!C:\Projects\Client\node_modules\css-loader\index.js!C:\Projects\Client\node_modules\sass-loader\index.js!C:\Projects\Client\assets\navigation.scss:152:48
        at module.exports (C:\Projects\Client\node_modules\style-loader\index.js!C:\Projects\Client\node_modules\css-loader\index.js!C:\Projects\Client\node_modules\sass-loader\index.js!C:\Projects\Client\assets\navigation.scss:174:69)
        at Object.<anonymous> (C:\Projects\Client\node_modules\style-loader\index.js!C:\Projects\Client\node_modules\css-loader\index.js!C:\Projects\Client\node_modules\sass-loader\index.js!C:\Projects\Client\assets\navigation.scss:54:37)
        at __webpack_require__ (C:\Projects\Client\node_modules\style-loader\index.js!C:\Projects\Client\node_modules\css-loader\index.js!C:\Projects\Client\node_modules\sass-loader\index.js!C:\Projects\Client\assets\navigation.scss:21:30)
        at C:\Projects\Client\node_modules\style-loader\index.js!C:\Projects\Client\node_modules\css-loader\index.js!C:\Projects\Client\node_modules\sass-loader\index.js!C:\Projects\Client\assets\navigation.scss:41:18
        at Object.<anonymous> (C:\Projects\Client\node_modules\style-loader\index.js!C:\Projects\Client\node_modules\css-loader\index.js!C:\Projects\Client\node_modules\sass-loader\index.js!C:\Projects\Client\assets\navigation.scss:44:10)

Also, I used the CLI to process my source file and it processed it with no errors reported and the output file was in the correct format.

**Comment by buddyackerman** [comment]
It gets even weirder.  I changed the loader to this:

    { test: /\.scss$/, loaders: ['style-loader', 'css-loader', 'sass-loader'] },

Now it generates the CSS properly and, of course, it's in the main js bundle.  Here's the weird thing, the bundle now ONLY has the css in it.  All of the script from my actual application is no longer in the bundle.  Here's the entry section of the config

    entry: {
        "app": './src/navigation.js',
        "styles": './src/styles.js'
    }

**Comment by evilebottnawi** [comment]
@buddyackerman don't use `extract-text-webpack-plugin` for `webpack@4` for extract css, use https://github.com/webpack-contrib/mini-css-extract-plugin. Thanks!

**Comment by buddyackerman** [comment]
@evilebottnawi I'm not using webpack 4.  I'm using webpack 1.13.2

**Comment by evilebottnawi** [comment]
@buddyackerman `webpack@1` and `webpack@2` are deprecated and no maintenance, please upgrade to `webpack@3`/`webpack@4` (latest is better)

**Comment by buddyackerman** [comment]
@evilebottnawi I can't because we still have to support IE8.

**Comment by evilebottnawi** [comment]
@buddyackerman I really would be happy to help, but it took a very long time and we have a lot of issues, i can't spent time on deprecation and unmaintenance versions, sorry. Try to debug and fix self, you have stack trace, just looks what is wrong and why

**Comment by buddyackerman** [comment]
@evilebottnawi If I knew anything about how webpack actually works or how loaders work then I might be able to make heads or tails of the stack trace but I don't so the stack track makes no sense. There is no such thing as `self` in a scss file and the line numbers given in the stac trace don't point to any line in the sass-loader that has anything to do with 'self'.  Even if they did it would still be pretty useless since I don't really have any way of running webpack/sass-loader in debug mode to find out what exactly is going on.

**Comment by evilebottnawi** [comment]
@buddyackerman prove your `style-loader` version?

**Comment by buddyackerman** [comment]
@evilebottnawi 

  "name": "style-loader",
  "version": "0.13.2",


  "name": "css-loader",
  "version": "0.25.0",

**Comment by evilebottnawi** [comment]
@buddyackerman can you update `style-loader` to latest (he should have compatibility with any `webpack` versions)

**Comment by buddyackerman** [comment]
@evilebottnawi After updating the style loader I got the following results:

Using  `{ test: /\.scss$/, loaders: ['style-loader', 'css-loader', 'sass-loader'] },`  I got the same css only output in the bundle, no other js script was processed


Using  `{ test: /\.scss$/, loader: ExtractTextPlugin.extract('style-loader', 'css-loader', 'sass-loader') },`  (no brackets for the loader array) I got an effectively empty output bundle, it included only the webpack bootstrap code.


Using  `{ test: /\.scss$/, loader: ExtractTextPlugin.extract(['style-loader', 'css-loader', 'sass-loader']) },` I got this error:

    ERROR in ./assets/navigation.scss
    Module build failed: ReferenceError: window is not defined
        at C:\Projects\navigation\node_modules\style-loader\index.js!C:\Projects\navigation\node_modules\css-loader\index.js!C:\Projects\navigation\node_modules\sass-loader\index.js!C:\Projects\navigation\assets\navigation.scss:190:3
        at C:\Projects\navigation\node_modules\style-loader\index.js!C:\Projects\navigation\node_modules\css-loader\index.js!C:\Projects\navigation\node_modules\sass-loader\index.js!C:\Projects\navigation\assets\navigation.scss:179:47
        at module.exports (C:\Projects\navigation\node_modules\style-loader\index.js!C:\Projects\navigation\node_modules\css-loader\index.js!C:\Projects\navigation\node_modules\sass-loader\index.js!C:\Projects\navigation\assets\navigation.scss:244:89)
        at Object.<anonymous> (C:\Projects\navigation\node_modules\style-loader\index.js!C:\Projects\navigation\node_modules\css-loader\index.js!C:\Projects\navigation\node_modules\sass-loader\index.js!C:\Projects\navigation\assets\navigation.scss:63:37)
        at __webpack_require__ (C:\Projects\navigation\node_modules\style-loader\index.js!C:\Projects\navigation\node_modules\css-loader\index.js!C:\Projects\navigation\node_modules\sass-loader\index.js!C:\Projects\navigation\assets\navigation.scss:21:30)
        at C:\Projects\navigation\node_modules\style-loader\index.js!C:\Projects\navigation\node_modules\css-loader\index.js!C:\Projects\navigation\node_modules\sass-loader\index.js!C:\Projects\navigation\assets\navigation.scss:41:18
        at Object.<anonymous> (C:\Projects\navigation\node_modules\style-loader\index.js!C:\Projects\navigation\node_modules\css-loader\index.js!C:\Projects\navigation\node_modules\sass-loader\index.js!C:\Projects\navigation\assets\navigation.scss:44:10)
        at Module._compile (module.js:660:30)
        at Object.exec (C:\Projects\navigation\node_modules\webpack-core\lib\NormalModuleMixin.js:88:7)
        at Object.<anonymous> (C:\Projects\navigation\node_modules\extract-text-webpack-plugin\loader.js:97:22)
        at Compiler.<anonymous> (C:\Projects\navigation\node_modules\webpack\lib\Compiler.js:214:10)
        at C:\Projects\navigation\node_modules\webpack\lib\Compiler.js:403:12
        at Compiler.next (C:\Projects\navigation\node_modules\tapable\lib\Tapable.js:67:11)
        at Compiler.<anonymous> (C:\Projects\navigation\node_modules\extract-text-webpack-plugin\loader.js:78:5)
        at Compiler.next (C:\Projects\navigation\node_modules\tapable\lib\Tapable.js:69:14)
        at Compiler.<anonymous> (C:\Projects\navigation\node_modules\webpack\lib\CachePlugin.js:40:4)
        at Compiler.applyPluginsAsync (C:\Projects\navigation\node_modules\tapable\lib\Tapable.js:71:13)
        at Compiler.<anonymous> (C:\Projects\navigation\node_modules\webpack\lib\Compiler.js:400:9)
        at Compilation.<anonymous> (C:\Projects\navigation\node_modules\webpack\lib\Compilation.js:577:13)
        at Compilation.applyPluginsAsync (C:\Projects\navigation\node_modules\tapable\lib\Tapable.js:60:69)
        at Compilation.<anonymous> (C:\Projects\navigation\node_modules\webpack\lib\Compilation.js:572:10)
        at Compilation.applyPluginsAsync (C:\Projects\navigation\node_modules\tapable\lib\Tapable.js:60:69)
        at Compilation.<anonymous> (C:\Projects\navigation\node_modules\webpack\lib\Compilation.js:567:9)
        at Compilation.applyPluginsAsync (C:\Projects\navigation\node_modules\tapable\lib\Tapable.js:60:69)
        at Compilation.<anonymous> (C:\Projects\navigation\node_modules\webpack\lib\Compilation.js:563:8)
        at Compilation.applyPluginsAsync (C:\Projects\navigation\node_modules\tapable\lib\Tapable.js:60:69)
        at Compilation.seal (C:\Projects\navigation\node_modules\webpack\lib\Compilation.js:525:7)
        at Compiler.<anonymous> (C:\Projects\navigation\node_modules\webpack\lib\Compiler.js:397:15)
        at C:\Projects\navigation\node_modules\tapable\lib\Tapable.js:103:11
        at Compilation.<anonymous> (C:\Projects\navigation\node_modules\webpack\lib\Compilation.js:445:10)

**Comment by evilebottnawi** [comment]
@buddyackerman you run code in browser env?

**Comment by buddyackerman** [comment]
@evilebottnawi  The resulting build is run in a browser, yes.  I'm actually building an AngularJS module for using in our Angular apps.

BTW, I had been using a very similar webpack config for another Angular app successfully although not using sass/scss.  It's only using the sass loader that is causing the problems.

**Comment by evilebottnawi** [comment]
@buddyackerman maybe you have very buggy and old `sass-loader` :disappointed:

**Comment by buddyackerman** [comment]
@evilebottnawi If you were asking if I ran the above in a browser, then no.  This was the output of just running webpack build.

**Comment by buddyackerman** [comment]
@evilebottnawi I installed the latest version of the sass-loader

**Comment by buddyackerman** [comment]
@evilebottnawi Actually the sass-loader is v4.1.1 (I was thinking of the style loader that I just updated)

**Comment by buddyackerman** [comment]
@evilebottnawi I'd prefer that the build produces two files the main JS file and a CSS file but I'd be satisfied with a single JS bundle that included the css module.  However, even not using the extract text plugin only builds the CSS.  Why would it not bundle all the JS from my app also?  I would expect that at least that would work.

**Comment by hanyulo** [comment]
@buddyackerman I bump in to the same problem as you. Any suggestion for me to solve the issue? 
Here is my syntax: 
```
module: {
  loaders: [
    {
      test: /\.scss$/,
      exclude: /node_modules/,
      loader: ExtractTextPlugin.extract('style-loader', 'css-loader', 'postcss-loader', 'sass-loader')
    },
  ]
},
```
* version of packages
  * "extract-text-webpack-plugin": "^1.0.1",
  * "webpack": "^1.13.3",
  * "sass-loader": "^4.1.1",
  * "node-sass": "^4.9.0",
  * "css-loader": "^0.26.1",

I would really appreciate that you can give me any related advice.

**Comment by buddyackerman** [comment]
@hanyulo Unfortunately I didn't get it fixed with webpack version 1.13, I wound up just upgrading to 3.10.  Fortunately the IE8 requirement finally went away so that I could use a newer version of webpack.

**Comment by chosenvictim** [comment]
I am getting exact same issue with webpack@1.13.2
Is there any solution ?
